### PR TITLE
Add TrackCatalogIndexReadDuration and TrackCatalogIndexWriteDuration

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/NuGetGallery/issues/5527

Depends on https://github.com/NuGet/NuGet.Services.Metadata/pull/269.

### CODE REVIEW

File: src/Ng/Program.cs [hunk 1]
```diff
@@ -47,6 +46,14 @@ public static void Main(string[] args)
                 ServicePointManager.SecurityProtocol &= ~SecurityProtocolType.Ssl3;
                 ServicePointManager.SecurityProtocol |= SecurityProtocolType.Tls12;
 
+                // Determine the job name
+                if (args.Length == 0)
+                {
+                    throw new ArgumentException("Missing tool specification");
```
Comment (loic-sharma) [inline src/Ng/Program.cs]: >tool specification [](start = 57, length = 18)

Maybe this would be an better error message: "Missing job name argument"?

File: src/Ng/Program.cs [hunk 2]
```diff
@@ -59,17 +66,10 @@ public static void Main(string[] args)
...
-                if (args.Length == 0)
-                {
-                    throw new ArgumentException("Missing tool specification");
-                }
 
                 // Create an ITelemetryService
                 var telemetryService = new TelemetryService(new TelemetryClient());
 
-                var jobName = args[0];
-                LogContext.PushProperty("JobName", jobName);
```
Comment (loic-sharma) [inline src/Ng/Program.cs]: >LogContext [](start = 16, length = 10)

Why was this removed? Does it not work?

Comment (joelverhagen) [reply src/Ng/Program.cs]: Yes, it works, but only on traces. I added a more general solution for all AI events.

File: src/Ng/Program.cs (hunk 1 above)
Comment (joelverhagen) [reply src/Ng/Program.cs]: Sure

Comment (joelverhagen) [reply src/Ng/Program.cs]: Fixed.
